pkg/operator/configobserver/node/suppress_config_updates_test.go

```diff
@@ -221,18 +238,160 @@ func TestSuppressConfigUpdateUntilSameProfileFunc(t *testing.T) {
+				},
+			},
+			clusterProfile: configv1.LowUpdateSlowReaction,
+		},
+		{
+			name:                       "KCM: reject transition from Low to Default",
+			isSuppressionExpected:      true,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Low profile
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"5m0s"},
+				},
+			},
+			clusterProfile: configv1.DefaultUpdateDefaultReaction,
+		},
+		{
+			name:                       "KCM: reject transition from Empty to Low",
+			isSuppressionExpected:      true,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Empty profile
+				"extendedArguments": map[string]interface{}{},
+			},
+			clusterProfile: configv1.LowUpdateSlowReaction,
+		},
+		{
+			name:                       "KCM: reject transition from Low to Empty",
+			isSuppressionExpected:      true,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Low profile
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"5m0s"},
+				},
+			},
+			clusterProfile: "",
+		},
+		{
+			name:                       "KCM: should not reject transition from Default to Medium",
+			isSuppressionExpected:      false,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Default profile
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"40s"},
+				},
+			},
+			clusterProfile: configv1.MediumUpdateAverageReaction,
+		},
+		{
+			name:                       "KCM: should not reject transition from Medium to Default",
+			isSuppressionExpected:      false,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Medium profile
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"2m0s"},
+				},
+			},
+			clusterProfile: configv1.DefaultUpdateDefaultReaction,
+		},
+		{
+			name:                       "KCM: should not reject transition from Empty to Medium",
+			isSuppressionExpected:      false,
+			operatorLatencyConfigs:     kcmLatencyConfigs,
+			operatorRejectionScenarios: kcmRejectionScenarios,
+			operatorSpecObservedConfig: map[string]interface{}{
+				// Empty profile
+				"extendedArguments": map[string]interface{}{},
+			},
+			clusterProfile: configv1.MediumUpdateAverageReaction,
+		},
+		{
+			name:                       "KCM: should not reject transition from Medium to Empty",
```

Comment:
Medium to Low?

Reply:
Added Medium <-> Low in the test cases.
Also, added `latestAvailableRevision` in the test case struct to ensure that we can check the suppressor if it actually not suppress during revision-0 special case.